tests/measurements/test_measurements.py

```diff
@@ -0,0 +1,481 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the measurements module"""
+import numpy as np
+import pytest
+
+import pennylane as qml
+from pennylane.measurements import (
+    AllCounts,
+    Counts,
+    Expectation,
+    MeasurementProcess,
+    MeasurementValue,
+    MeasurementValueError,
+    MidMeasure,
+    MutualInfo,
+    Probability,
+    Sample,
+    Shadow,
+    ShadowExpval,
+    State,
+    Variance,
+    VnEntropy,
+    expval,
+    sample,
+    var,
+)
+from pennylane.operation import DecompositionUndefinedError
+from pennylane.queuing import AnnotatedQueue
+
+
+@pytest.mark.parametrize(
+    "return_type, value",
+    [
+        (Expectation, "expval"),
+        (Sample, "sample"),
+        (Counts, "counts"),
+        (Variance, "var"),
+        (Probability, "probs"),
+        (State, "state"),
+        (MidMeasure, "measure"),
+    ],
+)
+def test_ObservableReturnTypes(return_type, value):
+    """Test the ObservableReturnTypes enum value, repr, and enum membership."""
+
+    assert return_type.value == value
+    assert isinstance(return_type, qml.measurements.ObservableReturnTypes)
+    assert repr(return_type) == value
+
+
+def test_no_measure():
+    """Test that failing to specify a measurement
+    raises an exception"""
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.PauliY(0)
+
+    with pytest.raises(qml.QuantumFunctionError, match="must return either a single measurement"):
+        _ = circuit(0.65)
+
+
+def test_numeric_type_unrecognized_error():
+    """Test that querying the numeric type of a measurement process with an
+    unrecognized return type raises an error."""
+    mp = MeasurementProcess("NotValidReturnType")
+    with pytest.raises(qml.QuantumFunctionError, match="Cannot deduce the numeric type"):
+        mp.numeric_type()
+
+
+def test_shape_unrecognized_error():
+    """Test that querying the shape of a measurement process with an
+    unrecognized return type raises an error."""
+    mp = MeasurementProcess("NotValidReturnType")
+    with pytest.raises(qml.QuantumFunctionError, match="Cannot deduce the shape"):
+        mp.shape()
+
+
+class TestMeasurementValue:
```

Comment:
This class should be moved to `test_mid_measure.py`.

Reply:
Done!